### PR TITLE
Clean text fix

### PR DESCRIPTION
**Pull request by martineb**
Merging changes to:

training_data.py
labeled_line_sentence.py
model.py

The intent was to generalize some parts of the code to facilitate usage of the modules in a variety of scenarios.

### CODE REVIEW

File: utils/Utils.py [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+from nltk.stem.lancaster import LancasterStemmer
+
+class TextCleaner:
+    
+    stemmer = LancasterStemmer()
+
+    def __init__(self, translate_table):
+        self.translate_table = translate_table
+
+    def clean_text(self, text):
```
Comment (ldacosta) [inline utils/Utils.py]: I'd propose to explicitly indicate that this is a "real" function (in the mathematical sense), and that it returns a value without side effects; so `def clean_text(self, text) -> str:`

File: utils/Utils.py [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+        new_text = re.sub("([0-9]+)((,|\.)?[0-9]*)", " ||number|| ", text.lower().translate(self.translate_table))
+        return ' '.join([self.stemmer.stem(i) for i in nltk.word_tokenize(new_text)])
+
+    def clean_text_column(self, data, text_column):
```
Comment (ldacosta) [inline utils/Utils.py]: Needs documentation (ie, what does this do and why is it different from `clean_text` above?)

File: utils/doc2vec/model.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def __load__(self) -> bool:
 
     # #########
 
-    def __init__(self, docs_bodies: pd.DataFrame, window: int = 10, min_count: int = 2, size: int = 200):
+    def __init__(self, sentences: list, tags: list, window: int = 10, min_count: int = 2, size: int = 200):
```
Comment (ldacosta) [inline utils/doc2vec/model.py]: I like this, it's much clearer!!! Just a little detail: is the implicit assumption here that the 2 lists have same length and that `sentences[i]` has tag `tags[i]`?

**Comment by martineb** [comment]
I have adressed some of the concerns and documented the code. Hope this is sufficient for a merge!

**Comment by ldacosta** [comment]
+1

**Pull request merged**